### ISSUE TITLE
Cornice deserialization does not call .deserialize() on Colander schema

### ISSUE THREAD

**Issue by robvdl** [Cornices/cornice#357]
We have been using Colander + Cornice for some time, occasionally we run into the problem where we want to validate two fields together, currently that is a little hard because of how Cornice invokes Colander.

What really happens, is:

1. Cornice constructs the Colander schema class, \_\_init\_\_ is called in the Colander schema class.
2. Cornice loops over every field in the Colander schema and deserializes each field individually.

What would be way more ideal:

1. Cornice constructs the Colander schema class, \_\_init\_\_ is called in the Colander schema class
2. Cornice calls .deserialize() directly on the Colander class, allowing the .deserialize() method to be overridden if required.
3. The Colander.deserialize() method itself loops over the individual fields to deserialize them, NOT Cornice... this seems a bit .. wrong.

This becomes quite important if you want to override the deserialize method on the Colander schema class itself, currently, you can't do this.

**Comment by delijati** [comment]
There was already that problem look into comment in #340 and #330. As i proposed `colander` should be dependency and the self build ability with `type` should be removed.

**Comment by leplatrem** [comment]
The use-case is something like:

```py
    foobar = Service(name="foobar", path="/foobar")


    class FooBarSchema(MappingSchema):
        foo = SchemaNode(String(), type="str", missing=colander.drop)
        bar = SchemaNode(String(), type="str", missing=colander.drop)

        def deserialize(self, node, cstruct=null):
            appstruct = super(FooBarSchema, self).deserialize(node, cstruct)
            if 'foo' not in appstruct and 'bar' not in appstruct:
                message = "Must provide at least `foo` or `bar`"
                raise colander.Invalid(node, message, cstruct)
            return appstruct


    @foobar.post(schema=FooBarSchema)
    def foobar_post(request):
        return {"test": "succeeded"}
```

**Comment by leplatrem** [comment]
This is fixed in Cornice 2.0 (current master) \o/
